upgrade.sh

```diff
@@ -35,76 +18,134 @@ check_yq() {
   fi
 }
 
-check_new_version() {
-  if [[ "$CURRENT" == "$LATEST" ]]; then
-    echo -e "Docker snap is already updated\n"
-    exit 0
+check_multipass() {
+  if ! command -v multipass &>/dev/null; then
+    echo -e "multipass is not installed."
+    echo -e "Please install multipass first."
+    exit 1
   fi
 }
 
-main() {
-  check_yq
+launch_vm() {
+  echo "Launching Multipass VM..."
+  multipass launch noble -n "${VM_NAME}" -c 2 -m 4G -d 10G || {
```

Comment:
Should it check and remove VM_NAME first, it case the script has failed or interrupted?